ui/components/app/gas-customization/gas-modal-page-container/gas-modal-page-container.component.js

```diff
@@ -38,8 +41,23 @@ export default class GasModalPageContainer extends Component {
     customPriceIsExcessive: PropTypes.bool.isRequired,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      pollingToken: undefined,
+    };
+  }
+
   componentDidMount() {
-    this.props.fetchBasicGasEstimates();
+    getGasFeeEstimatesAndStartPolling().then((pollingToken) => {
+      this.setState({ pollingToken });
+    });
+  }
+
```

Comment:
Not sure if there is anything we can do about this:

Is it possible for the component to unmount before `getGasFeeEstimatesAndStartPolling()` resolves, thereby polling having been initiated in the controller but disconnect won't be called here, leaving the controller polling indefinitely?

Reply:
Hmmm.... good point. I think that affects our `useGasEstimates` hook, too. At least for this component, it won't matter long term because I plan on ceremoniously deleting it once @darkwing's efforts are fully realized. :D 

For the useGasEstimates hook we can do something like this 

```
useEffect(() => {
  let pollingToken;
  let isMounted = true;
  getGasFeeEstimatesAndStartPolling().then((token) => {
    if (!isMounted) {
      return disconnectMethodNameThatIForgot(token);
    }
    pollingToken = token;
  });
  return () => {
    isMounted = false;
    if (pollingToken) {
     //disconnect
    }
  }
}, []);
```